ocs_ci/ocs/resources/noobaa.py

```diff
@@ -0,0 +1,96 @@
+import base64
+import logging
+import boto3
+
+from ocs_ci.ocs.ocp import OCP
+
+logger = logging.getLogger(name=__file__)
+
+
+class NooBaa(object):
+    """
+    Wrapper class for NooBaa's S3 service
+    """
+
+    _s3_resource = None
+    _ocp_resource = None
+
+    def __init__(self):
+        """
+        Constructor for the NooBaa class
+        """
+        ocp_obj = OCP(kind='noobaa', namespace='noobaa')
+        results = ocp_obj.get()
+        endpoint = (
+            results.get('items')[0].get('status').get('services')
+            .get('serviceS3').get('externalDNS')[0]
+        )
+        creds_secret_name = (
+            results.get('items')[0].get('status').get('accounts')
+            .get('admin').get('secretRef').get('name')
+        )
+        secret_ocp_obj = OCP(kind='secret', namespace='noobaa')
+        results2 = secret_ocp_obj.get(creds_secret_name)
+
+        noobaa_access_key = base64.b64decode(
+            results2.get('data').get('AWS_ACCESS_KEY_ID')
+        ).decode('utf-8')
+        noobaa_secret_key = base64.b64decode(
+            results2.get('data').get('AWS_SECRET_ACCESS_KEY')
+        ).decode('utf-8')
+
+        self._ocp_resource = ocp_obj
```

Comment:
why not assigning this in line 22?

Reply:
I feel that assigning all the class members sequentially creates better readability and might save some confusion. Would you recommend assigning it in 22?

Reply:
ack